server/catgenome/src/main/java/com/epam/catgenome/util/MotifSearchIterator.java

```diff
@@ -52,22 +52,28 @@
     private final StrandSerializable strand;
     private final byte[] sequence;
     private final String regex;
+    private final int offset;
+    private final boolean includeSequence;
 
 
     public MotifSearchIterator(final byte[] seq, final String iupacRegex,
-                               final StrandSerializable strand, final String contig) {
-        if (strand != null && strand != StrandSerializable.POSITIVE && strand != StrandSerializable.NEGATIVE) {
+                               final StrandSerializable strand, final String contig,
+                               final int start, final boolean includeSequence) {
+        if (strand != null && strand != StrandSerializable.POSITIVE
+                && strand != StrandSerializable.NEGATIVE) {
             throw new IllegalStateException("Not supported strand: " + strand);
         }
         this.strand = strand;
         this.contig = contig;
         this.sequence = seq;
         this.regex = MotifSearcher.convertIupacToRegex(iupacRegex);
+        this.offset = start;
+        this.includeSequence = includeSequence;
         init();
     }
 
     private void init() {
-        if(strand == null) {
+        if (strand == null) {
```

Comment:
I would prefer if this method will be inlined in constructor and `polulateMatchers` actually would return list of matches that will assigned to  `positiveMatches` or `negativeMatches`. It is always better to return object from method rather then mutate existing one inside of the method